content/concepts/protocols.md

```diff
@@ -1,227 +1,149 @@
+or connection is called **protocol negotiation**.
 
-### Matching protocol ids and versions
+### Matching protocol IDs and versions
 
-When you register a protocol handler, there are two methods you can use.
+There are two methods available to register a protocol handler.
 
-The first takes two arguments: a protocol id, and a handler function. If an incoming stream request sends an exact
-match for the protocol id, the handler function will be invoked with the new stream as an argument.
+1. The first takes two arguments: a protocol ID and a handler function. If an incoming 
+stream request sends an exact match for the protocol ID; the handler function invokes 
+the new stream as an argument.
 
-#### Using a match function
-
-The second kind of protocol registration takes three arguments: the protocol id, a protocol match function, and the handler function.
-
-When a stream request comes in whose protocol id doesn't have any exact matches, the protocol id will be passed through
-all of the registered match functions. If any returns `true`, the associated handler function will be invoked.
-
-This gives you a lot of flexibility to do your own "fuzzy matching" and define whatever rules for protocol matching
-make sense for your application.
+2. **Using a match function**: The second kind of protocol registration takes three arguments: 
+   the protocol ID, a protocol match function, and the handler function.
+   - When a stream request comes in whose protocol ID doesn't have any exact matches, 
+     the protocol ID will pass through all the registered match functions if any returns 
+     `true`, the associated handler function invokes.
+   - This offers a lot of flexibility to do your own "fuzzy matching" and define 
+     whatever rules for protocol matching make sense of your application.
 
 ### Dialing a specific protocol
 
-When dialing a remote peer to open a new stream, the initiating peer sends the protocol id that they'd like to use. The remote peer will use
-the matching logic described above to accept or reject the protocol. If the protocol is rejected, the dialing peer can try again.
+When dialing a remote peer to open a new stream, the initiating peer sends the protocol 
+ID they'd like to use. The remote peer will use the matching logic described above to 
+accept or reject the protocol. If the protocol is rejected, the dialing peer can try again.
 
-When dialing, you can optionally provide a list of protocol ids instead of a single id. When you provide multiple protocol ids, they will
-each be tried in succession, and the first successful match will be used if at least one of the protocols is supported by the remote peer.
-This can be useful if you support a range of protocol versions, since you can propose the most recent version and fallback to older versions
-if the remote hasn't adopted the latest version yet.
+When dialing, you can optionally provide a list of protocol IDs instead of a single ID. 
```

Comment:
```suggestion
When opening a new stream, the initiator can optionally provide a list of protocol IDs instead of a single ID. 
```

plus other occurrences in the rest of the paragraph